exquery-restxq-specification/restxq-2.0-specification.html

```diff
@@ -646,6 +650,9 @@ <h3>Serialization</h3>
                   an implementation MUST throw an error. This is known as the
                   <dfn>Typed Resource Response rule</dfn>.
                 </li>
+                <li>
+                  If the result type is <code>multipart/*</code>, the function MAY return an array which is to be interpreted as the body parts of the sequence.  Each member of a multipart response MUST obey the <a>Typed Resource Response rule</a>, and MAY itself represent a multipart response with or without relevant HTTP headers.
```

Comment:
> MAY return an array which is to be interpreted as the body parts of the sequence

What is the structure/format of data in such an array please? Also I would like to understand why an array is required over a sequence?

Reply:
> I would like to understand why an array is required over a sequence?

There are a couple of reasons why we ought (IMO) to use an array rather than a sequence:

1. Multipart body parts may themselves have multipart types.  This is particularly true for `multipart/digest`, but could be true for most subtypes.  This is not possible in a sequence because of nesting rules.
2. If we allow associating multipart body parts to resource function parameters, it should be possible for those parameters to represent sequences.  Again, this is not possible in a sequence because of nesting rules.

It is easy enough to convert an array (even a nested one) into a sequence, which is handy for backwards compatibility with e.g. BaseX's approach to multipart, but doing so does potentially destroy nesting information.

Reply:
> > MAY return an array which is to be interpreted as the body parts of the sequence
> 
> What is the structure/format of data in such an array please?

My intention was that each member of that array would obey the same rules as if it were a singleton returned resource (with or without headers).

So each member of a multipart array would either be the resource to be returned (implied/no headers), or have the `item()+` result type where the first item in the result sequence is the `rest:response` document node, and the following item is the resource itself.

I'll have a go at making this more explicit.

Reply:
Sorry for jumping in: As the nesting of multipart bodies has been deprecated some time ago (see e.g. https://datatracker.ietf.org/doc/html/rfc7578#section-4.3), I wonder whether we should still support them? I haven’t seen them in the wild, but maybe others have?